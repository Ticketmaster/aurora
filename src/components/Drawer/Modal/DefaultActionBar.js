import React from "react";
import PropTypes from "prop-types";

import { colors } from "../../theme/index";

import CrossIcon from "../Icons/Cross";

import { withModal } from "./context";

import { CloseButtonContainer } from "./DefaultActionBar.styles";

// TODO: allow to set up the title and background
export const ModalActionBar = ({ modal: { closeModal }, iconButtonProps }) => (
  <CloseButtonContainer>
    <CrossIcon
      size={12}
      style={{ pointerEvent: "none" }}
      color={colors.onyx.base}
      className="button--close"
      aria-label="Close Modal"
      role="button"
      onClick={closeModal}
      {...iconButtonProps}
    />
  </CloseButtonContainer>
);

ModalActionBar.propTypes = {
  modal: PropTypes.shape({
    closeModal: PropTypes.func.isRequired
  }).isRequired,
  // eslint-disable-next-line react/forbid-prop-types
  iconButtonProps: PropTypes.object
};

ModalActionBar.defaultProps = {
  iconButtonProps: {}
};

export default withModal(ModalActionBar);
