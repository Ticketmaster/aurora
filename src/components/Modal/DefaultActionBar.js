/* eslint-disable import/prefer-default-export */
import React from "react";
import PropTypes from "prop-types";

import { colors } from "../../theme/index";

import CrossIcon from "../Icons/Cross";

import { withModal } from "./context";

import { CloseButtonContainer } from "./DefaultActionBar.styles";

export const ModalActionBar = ({ modal: { closeModal }, iconButtonProps }) => (
  <CloseButtonContainer>
    <CrossIcon
      size={12}
      style={{ pointerEvent: "none" }}
      color={colors.onyx.base}
      className="button--close"
      aria-label="Close Modal"
      role="button"
      onClick={closeModal}
      {...iconButtonProps}
    />
  </CloseButtonContainer>
);

ModalActionBar.propTypes = {
  modal: PropTypes.shape({
    closeModal: PropTypes.func.isRequired
  }).isRequired,
  // eslint-disable-next-line react/forbid-prop-types
  iconButtonProps: PropTypes.object
};

ModalActionBar.defaultProps = {
  iconButtonProps: {}
};

export default withModal(ModalActionBar);
