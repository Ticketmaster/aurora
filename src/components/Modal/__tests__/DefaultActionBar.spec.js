import React from "react";
import { render, Simulate } from "react-testing-library";

import { ModalActionBar } from "../DefaultActionBar";

describe("<DefaultActionBar />", () => {
  it("should render without errors", () => {
    const modal = {
      closeModal: () => {}
    };

    const { container } = render(
      <ModalActionBar
        iconButtonProps={{ "data-testid": "close-button" }}
        modal={modal}
      />
    );

    expect(container).toMatchSnapshot();
  });

  it("should call closeModal on the close button click", () => {
    const clickSpy = jest.fn();

    const modal = {
      closeModal: clickSpy
    };

    const { getByTestId } = render(
      <ModalActionBar
        iconButtonProps={{ "data-testid": "close-button" }}
        modal={modal}
      />
    );

    Simulate.click(getByTestId("close-button"));

    expect(clickSpy).toHaveBeenCalled();
  });
});
