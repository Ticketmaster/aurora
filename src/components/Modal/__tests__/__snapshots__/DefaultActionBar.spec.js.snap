// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DefaultActionBar /> should render without errors 1`] = `
<div>
  <div
    class="sc-bdVaJa gHhWxS"
  >
    <svg
      aria-label="Close Modal"
      class="button--close"
      data-testid="close-button"
      fill="rgba(38, 38, 38, 1)"
      height="12"
      role="button"
      viewBox="0 0 12 12"
      width="12"
    >
      <path
        d="M6.563 6.203l4.523 4.516a.384.384 0 0 1 .117.281c0 .11-.039.203-.117.281a.378.378 0 0 1-.137.09.417.417 0 0 1-.297 0 .378.378 0 0 1-.136-.09L6 6.766 1.484 11.28a.378.378 0 0 1-.136.09.417.417 0 0 1-.297 0 .378.378 0 0 1-.137-.09A.384.384 0 0 1 .797 11c0-.11.039-.203.117-.281l4.524-4.516L.913 1.68a.384.384 0 0 1-.117-.282c0-.109.039-.203.117-.28A.388.388 0 0 1 1.2 1c.112 0 .207.04.285.117L6 5.633l4.516-4.516A.388.388 0 0 1 10.8 1c.112 0 .207.04.285.117a.384.384 0 0 1 .117.281c0 .11-.039.204-.117.282L6.562 6.203z"
      />
    </svg>
  </div>
</div>
`;
